packages/dds/tree/src/shared-tree/schematizeTree.ts

```diff
@@ -223,7 +223,7 @@ export function ensureSchema(
 			return false;
 		}
 		case UpdateType.SchemaCompatible: {
-			checkout.updateSchema(viewSchema.viewSchemaAsStored);
+			checkout.updateSchema(toStoredSchema(viewSchema.viewSchemaRoot));
```

Comment:
Now that we are no longer using stored schema vs stored schema comparisons as part of the checks before updating, I think it may make sense to add an additional check that the result of toStoredSchema is a superset of the old schema (using the existing code we have for that).

Such a check could detect bugs in the schema compatibility checker or toStored schema which could lead to document corruption otherwise.

Adding such an assert inside of TreeCheckout.updateSchema would be a good defensive measure. Since your new checks more strict that allowsRepoSuperset, we should not get any false positives (asserts when it should be fine)